bot/cogs/moderation/utils.py

```diff
@@ -25,30 +24,42 @@
 RULES_URL = "https://pythondiscord.com/pages/rules"
 APPEALABLE_INFRACTIONS = ("ban", "mute")
 
-UserTypes = t.Union[discord.Member, discord.User]
+UserTypes = t.Union[discord.Member, discord.User, FetchedUser]
 MemberObject = t.Union[UserTypes, discord.Object]
 Infraction = t.Dict[str, t.Union[str, int, bool]]
 Expiry = t.Union[Duration, ISODateTime]
 
 
-def proxy_user(user_id: str) -> discord.Object:
+async def post_user(ctx: Context, user: t.Union[discord.User, discord.Object]) -> t.Optional[dict]:
     """
-    Create a proxy user object from the given id.
+    Create a new user in the database.
 
-    Used when a Member or User object cannot be resolved.
+    Used when an infraction needs to be applied on a user absent in the guild.
     """
-    log.trace(f"Attempting to create a proxy user for the user id {user_id}.")
+    log.trace("Attempting to add user to the database.")
 
-    try:
-        user_id = int(user_id)
-    except ValueError:
-        raise commands.BadArgument
+    if not isinstance(user, discord.User):
```

Comment:
This won't catch `discord.Member`.
```suggestion
    if not isinstance(user, (discord.Member, discord.User)):
```